src/resources/addys/claimables/utils.ts

```diff
@@ -1,17 +1,50 @@
 import { NativeCurrencyKey } from '@/entities';
 import { AddysClaimable, Claimable } from './types';
-import { convertRawAmountToBalance, convertRawAmountToNativeDisplay, greaterThan, lessThan } from '@/helpers/utilities';
+import { convertRawAmountToBalance, convertRawAmountToNativeDisplay, greaterThan } from '@/helpers/utilities';
 
 export const parseClaimables = (claimables: AddysClaimable[], currency: NativeCurrencyKey): Claimable[] => {
   return claimables
-    .map(claimable => ({
-      name: claimable.name,
-      uniqueId: claimable.unique_id,
-      iconUrl: claimable.dapp.icon_url,
-      value: {
-        claimAsset: convertRawAmountToBalance(claimable.amount, claimable.asset),
-        nativeAsset: convertRawAmountToNativeDisplay(claimable.amount, claimable.asset.decimals, claimable.asset.price.value, currency),
-      },
-    }))
+    .map(claimable => {
```

Comment:
It'd be more efficient to do one for...of loop where you sort, filter, and parse instead of mapping, filtering, and then sorting all in different iterations.

Reply:
true but users will only have a handful of claimables so doesn't matter too much